src/Ansatz/Chain.jl

```diff
@@ -144,6 +146,118 @@ function rightindex(::Union{Open,Periodic}, tn::Chain, site::Site)
     end
 end
 
+"""
+    adjoint(tn::TensorNetwork{<:Quantum})
+
+Return the adjoint [`TensorNetwork`](@ref).
+
+# Implementation details
+
+The tensors are not transposed, just `conj!` is applied to them.
+"""
+function Base.adjoint(qtn::Chain)
+    qtn = deepcopy(qtn)
+
+    foreach(conj!, tensors(qtn))
+
+    return qtn
+end
```

Comment:
`Base.adjoint` already exists for `Quantum`. You should use that inside and implement it for all `Ansatz`, not only `Chain`.